snapcraft/plugins/go.py

```diff
@@ -122,30 +115,25 @@ def clean_pull(self):
         if os.path.exists(self._gopath):
             shutil.rmtree(self._gopath)
 
-    def _local_pull(self):
+    def _get_local_go_package(self):
         if self.options.go_importpath:
             go_package = self.options.go_importpath
         else:
             logger.warning(
                 'Please consider setting `go-importpath` for the {!r} '
                 'part'.format(self.name))
-            go_package = os.path.basename(
-               os.path.abspath(self.options.source))
-            local_path = os.path.join(self._gopath_src, go_package)
-            if os.path.islink(local_path):
-                os.unlink(local_path)
-            os.symlink(self.sourcedir, local_path)
-        self._run(['go', 'get', '-t', '-d', './{}/...'.format(go_package)])
-
-    def _remote_pull(self):
-        for go_package in self.options.go_packages:
-            self._run(['go', 'get', '-t', '-d', go_package])
+            go_package = os.path.basename(os.path.abspath(self.options.source))
+            go_package = go_package.rstrip('.git')
+        return go_package
 
     def build(self):
         super().build()
-        if self.options.source is not None:
-            self._local_build()
-        self._remote_build()
+
+        for go_package in self.options.go_packages:
+            self._run(['go', 'install', go_package])
+        if not self.options.go_packages:
+            self._run(['go', 'install',
+                       './{}/...'.format(self._get_local_go_package())])
```

Comment:
El 04/08/16 a las 19:46, Leo Arias escribió:
> In snapcraft/plugins/go.py
> <https://github.com/snapcore/snapcraft/pull/709#discussion_r73616018>:
>
> >  
> >      def build(self):
> >          super().build()
> > -        if self.options.source is not None:
> > -            self._local_build()
> > -        self._remote_build()
> > +
> > +        for go_package in self.options.go_packages:
> > +            self._run(['go', 'install', go_package])
> > +        if not self.options.go_packages:
> > +            self._run(['go', 'install',
> > +                       './{}/...'.format(self._get_local_go_package())])
>
> Because there is a windows vendored dependency inside that is missing
> the buildtag. It tries to compile it and fails.

Yeah, but build tags would solve this as it would having a:
os_linux.go
os_windows.go
os_macos.go

in the same package :-)